sled-agent/src/opte/illumos/port_manager.rs

```diff
@@ -267,21 +264,28 @@ impl PortManager {
         // any of the remaining fallible operations fail.
         let port_name = self.inner.next_port_name();
         let hdl = OpteHdl::open(OpteHdl::DLD_CTL)?;
-        hdl.create_xde(
-            &port_name,
-            MacAddr::from(mac.into_array()),
-            private_ip,
-            opte_subnet,
-            MacAddr::from(gateway.mac.into_array()),
-            gateway_ip,
-            boundary_services.ip,
-            boundary_services.vni,
+
+        // TODO-completeness: Add support for IPv6.
+        let vpc_cfg = VpcCfg {
+            ip_cfg: IpCfg::Ipv4(Ipv4Cfg {
+                vpc_subnet: opte_subnet,
+                private_ip,
+                gateway_ip,
+                snat,
+                external_ips: external_ip,
+            }),
+            private_mac: MacAddr::from(mac.into_array()),
+            gateway_mac: MacAddr::from(gateway.mac.into_array()),
             vni,
-            self.inner.underlay_ip,
-            snat,
-            external_ip,
-            /* passthru = */ false,
-        )?;
+            phys_ip: self.inner.underlay_ip.into(),
+            boundary_services,
+            // TODO-remove: Part of the external IP hack.
+            proxy_arp_enable: true,
```

Comment:
Hmmmm, should this be `true` by default? I don't know if that's equiv to how it was before. I'm concerned this might break the work @rcgoodfellow has been doing.

Reply:
Fair question. I think this field is actually irrelevant. [In `xde` itself](https://github.com/oxidecomputer/opte/blob/master/xde/src/xde.rs#L497), we check the `xde_ext_ip_hack` flag, and use _that_ to determine if we're doing proxy ARP. If that's true, we copy the rest of the VPC configuration.

It looks kind of like this is an annoying side-effect of using the `VpcCfg` both externally and inside `xde` itself. We may actually just want to remove this flag from the `VpcCfg`. It looks like `xde` only refers to it in that one place, and otherwise directly checks the `xde_ext_ip_hack` flag. Thoughts?

Reply:
We can't remove it without doing more work because we need to check for it in various parts of the VPC impl (I really cannot wait to rip this hack out). But looking at xde it looks like it will always override the value passed down anyways. So it seems it should be irrelevant. But we should just add a comment that xde will override whatever value we place here, because god knows how much longer this hack is going to persist and I don't want people wasting time trying to figure out this mess.

Reply:
Cool, comment it is. Thanks.